cognite/client/data_classes/data_modeling/instances.py

```diff
@@ -685,6 +689,10 @@ def as_ids(self) -> list[NodeId]:
 class NodeList(CogniteResourceList[Node]):
     _RESOURCE = Node
 
+    def __init__(self, resources: Collection[Any], cognite_client: CogniteClient = None):
+        super().__init__(resources, cognite_client)
+        self.cursor: str | None = None
```

Comment:
Let's create dedicated `NodeListWithCursor(NodeList)` and `EdgeListWithCursor` classes for these.